### PR TITLE
Implement a mechanism for embedding YAML addons into KubeOne binary

### LINKED ISSUE #1351: Implement a mechanism for embedding YAML addons into KubeOne binary

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

This PR implements a mechanism for embedding YAML addons into the KubeOne binary.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1351

**Does this PR introduce a user-facing change?**:
```release-note
Implement a mechanism for embedding YAML addons into KubeOne binary
```

/assign @kron4eg

### CODE REVIEW

File: pkg/addons/ensure.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+		return err
+	}
+	if applier.LocalFS == nil {
+		s.Logger.Infoln("Skipping applying addons because addons are not enabled...")
+		return nil
+	}
+
+	s.Logger.Infof("Applying user provided addons...")
+
+	addons, err := fs.ReadDir(applier.LocalFS, ".")
```
Comment (xmudrii) [inline pkg/addons/ensure.go:58]: Check is `LocalFS != nil`.

**Review by kron4eg — APPROVED** [review]

**Pull request merged**